cmd/node.go

```diff
@@ -0,0 +1,609 @@
+package cmd
+
+import (
+	"fmt"
+	"os"
+	"os/exec"
+	"runtime"
+	"strings"
+	"syscall"
+	"time"
+
+	"github.com/algorandfoundation/hack-tui/ui/style"
+	"github.com/spf13/cobra"
+)
+
+var nodeCmd = &cobra.Command{
+	Use:   "node",
+	Short: "Algod installation",
+	Long:  style.Purple(BANNER) + "\n" + style.LightBlue("View the node status"),
+}
+
+var installCmd = &cobra.Command{
+	Use:   "install",
+	Short: "Install Algod",
+	Long:  "Install Algod on your system",
+	Run: func(cmd *cobra.Command, args []string) {
+		installNode()
+	},
+}
+
+var configureCmd = &cobra.Command{
+	Use:   "configure",
+	Short: "Configure Algod",
+	Long:  "Configure Algod settings",
+	Run: func(cmd *cobra.Command, args []string) {
+		configureNode()
+	},
+}
+
+var startCmd = &cobra.Command{
+	Use:   "start",
+	Short: "Start Algod",
+	Long:  "Start Algod on your system (the one on your PATH).",
+	Run: func(cmd *cobra.Command, args []string) {
+		startNode()
+	},
+}
+
+var stopCmd = &cobra.Command{
+	Use:   "stop",
+	Short: "Stop Algod",
+	Long:  "Stop the Algod process on your system.",
+	Run: func(cmd *cobra.Command, args []string) {
+		stopNode()
+	},
+}
+
+var upgradeCmd = &cobra.Command{
+	Use:   "upgrade",
+	Short: "Upgrade Algod",
+	Long:  "Upgrade Algod (if installed with package manager).",
+	Run: func(cmd *cobra.Command, args []string) {
+		upgradeAlgod()
+	},
+}
+
+func init() {
+	rootCmd.AddCommand(nodeCmd)
+	nodeCmd.AddCommand(installCmd)
+	nodeCmd.AddCommand(configureCmd)
+	nodeCmd.AddCommand(startCmd)
+	nodeCmd.AddCommand(stopCmd)
+	nodeCmd.AddCommand(upgradeCmd)
+}
+
+func installNode() {
+	fmt.Println("Checking if Algod is installed...")
+
+	// Check if Algod is installed
+	if !isAlgodInstalled() {
+		fmt.Println("Algod is not installed. Installing...")
+
+		// Install Algod based on OS
+		switch runtime.GOOS {
+		case "linux":
+			installNodeLinux()
+		case "darwin":
+			installNodeMac()
+		default:
+			panic("Unsupported OS: " + runtime.GOOS)
+		}
+	} else {
+		fmt.Println("Algod is already installed.")
+		printAlgodInfo()
+	}
+
+}
+
+func installNodeLinux() {
+	fmt.Println("Installing Algod on Linux")
+
+	// Check that we are calling with sudo
+	if !isRunningWithSudo() {
+		fmt.Println("This command must be run with super-user priviledges (sudo).")
+		os.Exit(1)
+	}
+
+	var installCmds [][]string
+	var postInstallHint string
+
+	// Based off of https://developer.algorand.org/docs/run-a-node/setup/install/#installation-with-a-package-manager
+
+	if checkCmdToolExists("apt") { // On Ubuntu and Debian we use the apt package manager
```

Comment:
Probably should only use apt-get since `apt` is considered unstable. At least it still warns in the terminal when using it